Logging format cstring cleanup

JCantu248 opened the pull request:
## 🗣 Description ##

The changes on this branch involve changing all the f-strings and .format style strings in logging messages to the C style string format. 

A small tweak was also made on `gophish-export.py` line 368-369 to change from `campaign_id` to `campaign["id"]`, which will improve readability of the message. `campaign_id` is a Gophish generated integer, which does not really work well with the rest of the statement. `campaign["id"]` is the actual campaign name sourced from Gophish. 

## 💭 Motivation and context ##

This change is necessary to bring gophish-tools in line with the rest of the best practices, which include using the C-style format strings for Python logging. The repository source code was using a mixture of f-strings, .format, and C-style formatting in logging messages. Now the repository will have more consistency in this regard. 

## 🧪 Testing ##

I ran Pytest just to make sure everything still works as intended, and received all green lights. I tested the changes by importing them into the COOL environment and running the LiPCA Setup script to make sure that logging was still working as intended. 

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

JCantu248 committed:
Improvement branch for https://github.com/cisagov/gophish-tools/issues/34

Combed through the code for every logger to have f-strings and .format statements converted to c-strings. C style formatting is the preferred standard for use with logging statements.

JCantu248 committed:
Bump project version from 0.0.6 to 0.0.7

JCantu248 committed:
Testing that logger output still works with current format changes.

This will be tested in the COOL against current projects.

JCantu248 committed:
Replaced campaign.name with campaign["id"]

Campaign at line 369 is not the Gophish API campaign object, but a dict with the name valued stored with the key value "id"

JCantu248 committed:
Bump project version from 0.0.7-rc.1 to 0.0.7-rc.2

JCantu248 committed:
Bump project version from 0.0.7-rc.2 to 0.0.7

jsf9k commented on src/tools/gophish_cleaner.py:
```diff
@@ -160,17 +159,16 @@ def main() -> None:
-            '"{}"is not a valid logging level.  Possible values are debug, info, warning, and error.'.format(
-                log_level
-            )
+            '"%s"is not a valid logging level.  Possible values are debug, info, warning, and error.',
```
Do you need an extra space here?
```suggestion
            '"%s" is not a valid logging level.  Possible values are debug, info, warning, and error.',
```

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_complete.py:
```diff
@@ -167,14 +167,15 @@ def main() -> None:
     except ValueError:
         logging.critical(
-            f'"{log_level}"is not a valid logging level. Possible values are debug, info, warning, and error.'
+            '"%s"is not a valid logging level. Possible values are debug, info, warning, and error.',
```
I _think_ you need an extra space here.
```suggestion
            '"%s" is not a valid logging level. Possible values are debug, info, warning, and error.',
```

JCantu248 replied:
See 825962b

jsf9k commented on src/tools/gophish_export.py:
```diff
@@ -382,15 +386,16 @@ def main() -> None:
     except ValueError:
         logging.critical(
-            f'"{log_level}"is not a valid logging level. Possible values are debug, info, warning, and error.'
+            '"%s"is not a valid logging level. Possible values are debug, info, warning, and error.',
```
Extra space again.
```suggestion
            '"%s" is not a valid logging level. Possible values are debug, info, warning, and error.',
```

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -53,10 +53,10 @@ def load_landings(api, assessment):
+        logging.debug("Page Name: %s", new_page.name)
+        logging.debug("Redirect ULR: %s", new_page.redirect_url)
+        logging.debug("Capture Credentials: %s", new_page.capture_credentials)
+        logging.debug("Capture Passwords: %s", new_page.capture_passwords)
```
Please take care of the code scanning error here.  If it's a false positive then please document that with a comment.

JCantu248 replied:
What kind of code scanning error are you encountering here?

We just used this branch for a demo, and it worked as we expected. I havent seen any irregularities from the editor either.

JCantu248 replied:
Ah okay, I found the issue. 

I dont know why the credentials were being logged like they were. I removed the offending lines for now, and if they become necessary at some point in the future we can revisit adding them in and figuring out some solution to log them in an encrypted manner. 

See a65955c

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -117,23 +117,24 @@ def load_groups(api, assessment):
                             logging.info("Re-Loading new group.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s\n", e)
```
Why is a trailing `\n` required here and just below?  I don't think you do that anywhere else in the code.  I suggest removing it if there isn't a specific reason for it.

JCantu248 replied:
Im not sure why it was there in the first place, but it was removed 

See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -117,23 +117,24 @@ def load_groups(api, assessment):
         group["id"] = new_group.id
 
-        logging.info("Group Ready: {}\n".format(new_group.name))
+        logging.info("Group Ready: %s\n", new_group.name)
```
Why is a trailing `\n` required here and just above?  I don't think you do that anywhere else in the code.  I suggest removing it if there isn't a specific reason for it.

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -164,21 +165,22 @@ def build_campaigns(api, assessment):
                             logging.info("Re-Loading new template.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s\n", e.message)
```
Same question about the `\n` here.

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -202,29 +204,30 @@ def build_campaigns(api, assessment):
                             logging.info("Re-Loading new SMTP.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s\n", e.message)
```
`\n`

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -261,15 +264,14 @@ def main() -> None:
-            '"{}"is not a valid logging level.  Possible values are debug, info, warning, and error.'.format(
-                log_level
-            )
+            '"%s"is not a valid logging level.  Possible values are debug, info, warning, and error.',
```
I think you need a space here.
```suggestion
            '"%s" is not a valid logging level.  Possible values are debug, info, warning, and error.',
```

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -281,12 +283,12 @@ def main() -> None:
             assessment = json.load(json_file)
     except FileNotFoundError as e:
-        logging.error(f"{e}\n")
+        logging.error("%s\n", e)
```
`\n`

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -281,12 +283,12 @@ def main() -> None:
         sys.exit(1)
     except PermissionError as e:
-        logging.error(f"{e}\n")
+        logging.error("%s\n", e)
```
`\n`

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_test.py:
```diff
@@ -140,16 +140,15 @@ def main() -> None:
-            '"{}"is not a valid logging level.  Possible values are debug, info, warning, and error.'.format(
-                log_level
-            )
+            '"%s"is not a valid logging level.  Possible values are debug, info, warning, and error.',
```
Space.
```suggestion
            '"%s" is not a valid logging level.  Possible values are debug, info, warning, and error.',
```

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -69,19 +69,19 @@ def load_landings(api, assessment):
 
         # Returns Landing Page ID
-        logging.info(f"Landing Page {new_page.name} loaded.\n")
+        logging.info("Landing Page %s loaded.\n", new_page.name)
```
I don't think the `\n` is needed here.

JCantu248 replied:
See 825962be74cfc038bdf689b711d0b91860d5c0b5

jsf9k reviewed (changes requested):
I requested a bunch of minor changes.  Feel free to tell me why they don't make sense if you disagree with them.

JCantu248 committed:
Code changes for feedback

Unnecessary newlines removed from logging messages, spaces where appropriate

JCantu248 committed:
Merge branch 'develop' of github.com:cisagov/gophish-tools into Logging_format_cstring_cleanup

JCantu248 committed:
Adding comment indicating there may be false errors or warnings for lines 57-60

JCantu248 committed:
Code scanning error resolved. Debug statements logging credentials and passwords were removed as they are unnecessary.

There is no need for credentials to be logged like this for debug purposes. If it turns out in the future they may be necessary, they can be added in as needed.

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -246,7 +248,7 @@ def build_campaigns(api, assessment):
             raise
 
-        logging.info(f"Campaign {campaign['name']} successfully loaded.\n")
+        logging.info("Campaign %s successfully loaded.\n", campaign["name"])
```
This line still has an unnecessary `\n`.

JCantu248 replied:
Its been removed. See 44e24edb9ac96eb42759bdea8a9229563811ca59

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -305,7 +306,7 @@ def main() -> None:
     except Exception as e:
-        logging.debug(f"{type(e)}: {e}")
+        logging.debug("%s: %s", type(e), e)
         logging.critical("Closing with an error. Assessment not successfully loaded.\n")
```
Another unnecessary `\n`.

JCantu248 replied:
See 44e24edb9ac96eb42759bdea8a9229563811ca59

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -305,7 +306,7 @@ def main() -> None:
 
     except Exception as e:
-        logging.debug(f"{type(e)}: {e}")
+        logging.debug("%s: %s", type(e), e)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
Yeah its been changed to use exception. Makes more sense in this context. 

See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Its also been updated here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -281,12 +282,12 @@ def main() -> None:
             assessment = json.load(json_file)
     except FileNotFoundError as e:
-        logging.error(f"{e}\n")
+        logging.error("%s", e)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
Yep, ```logging.exception()``` would be better off here, so its been changed 

See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Updated again here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -281,12 +282,12 @@ def main() -> None:
         sys.exit(1)
     except PermissionError as e:
-        logging.error(f"{e}\n")
+        logging.error("%s", e)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Also been updated here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -202,29 +203,30 @@ def build_campaigns(api, assessment):
                             logging.info("Re-Loading new SMTP.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s", e.message)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Also been updated here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -164,21 +164,22 @@ def build_campaigns(api, assessment):
                             logging.info("Re-Loading new template.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s", e.message)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Also been updated here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -117,23 +116,24 @@ def load_groups(api, assessment):
                             logging.info("Re-Loading new group.")
                 else:
-                    logging.error(f"{e}\n")
+                    logging.error("%s", e)
```
Why not use [`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) here instead of recreating the wheel?

JCantu248 replied:
See 44e24edb9ac96eb42759bdea8a9229563811ca59

JCantu248 replied:
Also been updated here 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k reviewed (changes requested):
Thanks for taking care of the issues I mentioned.  I noticed a couple of places that still have unnecessary `\n`s and I pointed out a handful of places where you could use `logging.exception()` instead of recreating the wheel.

The extra `\n`s should be fixed, but the `logging.exception()` jazz is just a suggestion.  If there is a reason why it doesn't make sense to use it just state it in the PR.

Re-request a review from me when you're done and I'll quickly re-review.

JCantu248 committed:
Removed more newlines and changed logging to use exception.

For logging.errors that were used in catch blocks, they have been revised to use the logging.exception function.

jsf9k commented on src/tools/gophish_import.py:
```diff
@@ -281,12 +282,12 @@ def main() -> None:
         sys.exit(1)
     except PermissionError as e:
-        logging.error(f"{e}\n")
+        logging.exception("%s", e)
```
Now that you are using `logging.exception()`, all the exception information will appear in the stack trace; hence, it would be better to do something like `logging.exception("Unable to write to assessment file")` (or whatever the message needs to be to be descriptive and useful) instead of simply `logging.exception("%s", e)`.

Just a suggestion that could be applied to most all of the `logging.exception()` calls.

JCantu248 replied:
I added some more verbose messaging to the exception. It should hopefully give a bit more guidance about what was happening prior to the exception. 

See 3ee423033fefa9873fd13190c0e6edb575cd0080

jsf9k reviewed (approved):
Approved, but please consider the suggestion I made.

JCantu248 committed:
Updated the logging exceptions to be a bit more verbose and helpful when producing messages

jsf9k reviewed (approved):
Nice work!

dav3r commented on src/tools/gophish_export.py:
```diff
@@ -132,7 +134,7 @@ def export_campaigns(api, assessment_id):
         campaigns.append(get_campaign_data(api, campaign_id))
 
-    logging.info(f"{len(campaigns)} campaigns found for assessment {assessment_id}.")
+    logging.info("%i campaigns found for assessment %s.", len(campaigns), assessment_id)
```
```suggestion
    logging.info("%d campaigns found for assessment %s.", len(campaigns), assessment_id)
```

JCantu248 replied:
Its been changed. 

See aa93585

dav3r replied:
👍  Thanks!

dav3r commented on src/tools/gophish_export.py:
```diff
@@ -98,7 +98,9 @@ def export_targets(api, assessment_id):
 
-    logging.info(f"{len(targets)} email targets found for assessment {assessment_id}.")
+    logging.info(
+        "%i email targets found for assessment %s.", len(targets), assessment_id
```
I believe `%d` is now more standard than `%i`, even though `%i` still works:
```suggestion
        "%d email targets found for assessment %s.", len(targets), assessment_id
```

I don't see `%i` in the docs anymore: https://docs.python.org/3/library/string.html#format-specification-mini-language

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -53,10 +53,9 @@ def load_landings(api, assessment):
-        logging.debug("Capture Passwords: {}".format(new_page.capture_passwords))
+
+        logging.debug("Page Name: %s", new_page.name)
+        logging.debug("Redirect ULR: %s", new_page.redirect_url)
```
Fix typo:
```suggestion
        logging.debug("Redirect URL: %s", new_page.redirect_url)
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -69,19 +68,19 @@ def load_landings(api, assessment):
                     for old_page in old_pages:
                         if old_page.name == new_page.name:
-                            logging.debug(f"Deleting Page with ID {old_page.id}")
+                            logging.debug("Deleting Page with ID %i", old_page.id)
```
Is the ID an integer (`%d`)?  Or is it a string (`%s`)?
```suggestion
                            logging.debug("Deleting Page with ID %d", old_page.id)
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -117,23 +116,26 @@ def load_groups(api, assessment):
                     groups = api.groups.get()
                     logging.debug(
-                        f"Checking {len(groups)} for previously imported group to get ID"
+                        "Checking %i for previously imported group to get ID",
```
```suggestion
                        "Checking %d for previously imported group to get ID",
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -117,23 +116,26 @@ def load_groups(api, assessment):
                     for old_group in groups:
                         if old_group.name == new_group.name:
-                            logging.debug(f"Deleting Group with ID {old_group.id}")
+                            logging.debug("Deleting Group with ID %i", old_group.id)
```
Same question as above about `%d` or `%s`:
```suggestion
                            logging.debug("Deleting Group with ID %d", old_group.id)
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -164,21 +166,25 @@ def build_campaigns(api, assessment):
                         if old_template.name == new_template.name:
                             logging.debug(
-                                f"Deleting Template with ID {old_template.id}"
+                                "Deleting Template with ID %i", old_template.id
```
Same question as above about `%d` or `%s`:
```suggestion
                                "Deleting Template with ID %d", old_template.id
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -202,29 +208,33 @@ def build_campaigns(api, assessment):
                     smtps = api.smtp.get()
                     logging.debug(
-                        f"Checking {len(smtps)} for previously imported smtp profiles to get ID"
+                        "Checking %i for previously imported smtp profiles to get ID",
```
```suggestion
                        "Checking %d for previously imported smtp profiles to get ID",
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -202,29 +208,33 @@ def build_campaigns(api, assessment):
                     for old_smtp in smtps:
                         if old_smtp.name == new_smtp.name:
-                            logging.debug(f"Deleting SMTP with ID {old_smtp.id}")
+                            logging.debug("Deleting SMTP with ID %i", old_smtp.id)
```
Same question as above about `%d` or `%s`:
```suggestion
                            logging.debug("Deleting SMTP with ID %d", old_smtp.id)
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_import.py:
```diff
@@ -202,29 +208,33 @@ def build_campaigns(api, assessment):
                 )
                 logging.warning(
-                    f"Previous Campaign with id {old_campaign.id} being deleted."
+                    "Previous Campaign with id %i being deleted.", old_campaign.id
```
Same question as above about `%d` or `%s`:
```suggestion
                    "Previous Campaign with id %d being deleted.", old_campaign.id
```

JCantu248 replied:
See aa93585

dav3r commented on src/tools/gophish_test.py:
```diff
@@ -54,9 +54,9 @@ def get_campaigns(api, assessment_id):
 
     # Sets err to true if assessmentCampaigns has 0 length.
-    logging.debug(f"Num Campaigns: {len(assessmentCampaigns)}")
+    logging.debug("Num Campaigns: %i", len(assessmentCampaigns))
```
```suggestion
    logging.debug("Num Campaigns: %d", len(assessmentCampaigns))
```

JCantu248 replied:
See aa93585

dav3r reviewed (changes requested):
Looks mostly good to me, but I have some questions and suggestions.

JCantu248 committed:
Response to feedback, changing C formatter from using %i to %d for integer values

dav3r reviewed (approved):
Solid work! 🏆
